cpp/src/io/comp/comp.cpp

```diff
@@ -397,8 +397,16 @@ void host_compress(compression_type compression,
                "Unsupported compression type: " + compression_type_name(compression));
   if (not host_compression_supported(compression)) { return false; }
   if (not device_compression_supported(compression)) { return true; }
-  // If both host and device compression are supported, use the host if the env var is set
-  return getenv_or("LIBCUDF_HOST_COMPRESSION", std::string{"OFF"}) == "ON";
+  // If both host and device compression are supported, dispatch based on the environment variable
+
+  auto const env_var = getenv_or("LIBCUDF_HOST_COMPRESSION", std::string{"OFF"});
+  if (env_var == "AUTO") {
+    constexpr auto const default_threshold = 128ul;
```

Comment:
Should the two default thresholds (for compression and decompression) be set in a common header file?